### PR TITLE
add option max_reflections_per_experiment to combine_experiments

### PR DESCRIPTION
**Pull request by dwpaley**
Analogous to min_reflections_per_experiment. Experiments are thrown out if they have more than the maximum reflections. This is meant to reject obvious multicrystal shots from xfel experiments, when combining expts after spotfinding.

### CODE REVIEW

File: command_line/combine_experiments.py [hunk 1]
```diff
@@ -523,7 +530,13 @@ def average_detectors(target, panelgroups, depth):
                     params.output.min_reflections_per_experiment is not None
                     and n_sub_ref < params.output.min_reflections_per_experiment
                 ):
-                    skipped_expts += 1
+                    skipped_expts_min_refl += 1
+                    continue
```
Comment (graeme-winter) [inline command_line/combine_experiments.py:534]: Code coverage report suggests that this is not tested currently - could be worth adding a test? There's probably plenty of useful example data in `dials_data` which could be used for this.

**Comment by dwpaley** [comment]
Thanks, Graeme. The application on my mind is for small-molecule still
shots where the biggest possible number of spots in a (single-crystalline)
frame is much smaller. So yes there's a range but if I see 100 spots I know
it can be rejected.

I added a test for min_refl and max_refl and will go ahead and merge.

Dan

**Comment by dwpaley** [comment]
Oops, I rebased this locally and pushed to master instead of merging this PR. @ndevenish please feel free to correct the history if needed. Let me know if you need anything from me. Thanks, Dan

**Comment by ndevenish** [comment]
Hi Daniel,

Since there was nothing more recent and noticed relatively quickly I've rolled it back so we can properly merge this. 

As Graeme alluded to, please add a `newsfragments/1369.feature` text file with a one-line description of the changes (this goes into the CHANGELOG). It looks like you should probably add yourself to the `AUTHORS` file also.

After that, feel free to merge the pull request (make sure to squash merge this!).

Thanks!

**Comment by dwpaley** [comment]
Thanks for the help! Sorry to cause trouble. I added the newsfragment and my name and squash merged this PR.
Best,
Dan

**Pull request merged**